crates/ruff_python_formatter/src/statement/stmt_delete.rs

```diff
@@ -1,12 +1,46 @@
-use crate::{not_yet_implemented, FormatNodeRule, PyFormatter};
-use ruff_formatter::{write, Buffer, FormatResult};
-use rustpython_parser::ast::StmtDelete;
+use crate::prelude::PyFormatContext;
+use crate::{expression::parentheses::Parenthesize, AsFormat, FormatNodeRule, PyFormatter};
+use ruff_formatter::prelude::{format_with, space, text, Formatter};
+use ruff_formatter::{write, Buffer, Format, FormatResult};
+use rustpython_parser::ast::{Expr, StmtDelete};
 
 #[derive(Default)]
 pub struct FormatStmtDelete;
 
 impl FormatNodeRule<StmtDelete> for FormatStmtDelete {
     fn fmt_fields(&self, item: &StmtDelete, f: &mut PyFormatter) -> FormatResult<()> {
-        write!(f, [not_yet_implemented(item)])
+        let StmtDelete { range: _, targets } = item;
+        write!(f, [text("del"), space(), DeleteList::new(targets)])
+    }
+}
+
+// TODO(cnpryer): Impl FormatRuleWithOptions Parenthesize
+#[derive(Debug)]
+struct DeleteList<'a> {
+    delete_list: &'a [Expr],
+}
+
+impl<'a> DeleteList<'a> {
+    const fn new(delete_list: &'a [Expr]) -> Self {
+        Self { delete_list }
+    }
+}
+
+impl Format<PyFormatContext<'_>> for DeleteList<'_> {
+    fn fmt(&self, f: &mut Formatter<PyFormatContext<'_>>) -> FormatResult<()> {
+        write!(
```

Comment:
you shouldn't need a write call here, the `JoinBuilder` writes internally (see `ExprSequence`)

Reply:
a98d7213c1e1c241ff9aeceb133c0552a8b19060

Reply:
I wanted to ask this earlier, so I apologize, but I spent some time actually reading [Python's AST docs](https://docs.python.org/3/library/ast.html#) to see if I'm conflating concepts.

My brain wants to interpret `del Expr*` as `del ExprSequence`. If that's the case, would we want to just reuse your `ExprSequence` implementation here? Maybe I'm missing something.

Reply:
See a1d98558ebe5ff452920bf74bfa3c040e32c7d23